### ISSUE TITLE
Sorting causes item deletion/duplication

### ISSUE THREAD

**Issue by ctstewart** [gr8pefish/IronBackpacks#133]
#### Issue Description:
Items are being deleted or duplicated when sorting your inventory with a backpack open. This occurs in the Direwolf20 1.10.2 modpack (despite your instructions, I was advised to submit this bug here).

#### What happens:
Items are duplicated or deleted, essentially putting the backpack back to the state that it was in before it was opened.

#### What you expected to happen:
I expected the items to get inserted or removed from the backpack as normal.

#### Steps to reproduce (important):

1. Open the backpack.
2. Add or remove items.
3. Middle-click to sort inventory.
4. Close inventory.
5. Re-open backpack. It is the same as it was before it was first opened, causing items to delete or duplicate.

____
#### Affected Versions (Do *not* use "latest"):

- IronBackpacks: 2.2.19
- Minecraft: 1.10.2
- Forge: 8.0.99.99

**Comment by gr8pefish** [comment]
Thanks for the bug report! I'm currently away from my computer, visiting family for Christmas, but I'll be sure to take a look at the issue when I return (just over a week from now).

Clarification question: step 3, is that sorting the player's inventory, or the backpack's inventory?

**Comment by ctstewart** [comment]
Thank you for the quick response! I should have clarified: if you sort the backpack's inventory, this issue does not happen, it is only when you sort the player's inventory. I can do my best to upload a video as well if that would help you.

**Comment by gr8pefish** [comment]
That would be great, if you don't mind!

**Comment by ctstewart** [comment]
No problem! I've never uploaded something to YouTube before, but I will do my best :) I actually just realized trying to duplicate it that this only happens on nested backpacks.

**Comment by ctstewart** [comment]
Here's the video: https://youtu.be/RbnTF7i2UlI. Let me know if you have any more questions about it! I love your mod, and I'm excited to see this bug fixed. I lost too many items before figuring out the cause :P

**Comment by gr8pefish** [comment]
Hmm, I can't replicate the issue. I am using Inventory Sorter (as that is what is in DW20), and Iron Backpacks, and I am not getting the same issue. [Video](https://streamable.com/s7nob)

Would it be possible for you to try and narrow down what mod may be causing this? Also, your Forge version listed above is not correct, make sure it is updated.

**Comment by gr8pefish** [comment]
Any update on this? I'm going to have to close the issue if I don't hear back from you.

**Comment by ctstewart** [comment]
I apologize for not getting back to you sooner. I sent Direwolf20 a message hoping that he would have some insight into this issue. He has not gotten back to me. I intend to do my own troubleshooting, but I am in the middle of a large move, so it might be a few days before I can do that. I would really appreciate it if you could keep this issue open for now.

**Comment by gr8pefish** [comment]
I wouldn't expect DW20 to respond, but I will wait and leave the issue up until you have time to try and duplicate it.

**Comment by darkmtbg2** [comment]
i seem to have hit the issue as well in the direwolf20 modpack(started the server with 1.2.1 and is on 1.3.2 now) as well i had an lumber axe (tinkers' contsruct) duped over a livingwood bow(Botania) enhanted. i usally use the "Middle click function sorting"(simple inventory sorting?). i might have lost some rings of magnetizing(Actually additions) because if it too. but i havent seen item dupes with that one just deleted at one time to another the issue. all of the items i possible could have lost to it seems to be items/blocks that carry Nbt data on them. or at leas those i have noticed

oh yeah its also a diamond back nested in a gold one(advanced nesting upgrade)

though the occurance seems to be really rare however.

Im running my world on a didicated server side
Dedicated server side:
- 1.10.2 Minecraft version
- 12.18.3.2185 Forge
- 2.2.21 iron backpacks 
- r.1.9-339 Botania
- 2.6.1 Tinkers' construct
- 0.10.2 +43 inventory sorter
- Actually Additions 1.10.2 -r97

Update i might have lost a the Ender sword(Ender io) highly enchanted

**Comment by babebridou** [comment]
Here's a "me too" on this issue. I'm also running DW20 modpack, single player, version 1.3.2 of the pack.

I lost quite a few bags to that bug. Here's how I reproduce it 100%:

in survival, spawn any non-leather bag - craft it or cheat it in, it isn't important. If you craft it, don't open it until you're done with the craft and tier upgrading, and still without opening it, add any upgrade. Put the bag into an inventory slot (not action bar, this is important) and right click it, it opens. Add an item. Close the bag by closing your inventory. Open the inventory, open the bag, the item disappeared. Any further item added to the bag will disappear, including other bags. When you shift-click the bag, it will keep saying "you must open this bag normally first" regardless of how many times you open it from your inventory.

The workaround: right after creating the bag and before adding any item, open it from your action bar to hear the "click" sound, and check if the bag can be renamed. Repeat until a shift-click lets you rename it. This initializes the bag and won't make your other bags go poof before you put them inside.

**Comment by gr8pefish** [comment]
Oh alright cool, so it is a duplicate of issue #88, thanks a lot for narrowing that down!

I'll look into fixing this bug in about a week. Unfortunately I am currently in a very intensive medical training course that leaves no free time, so this fix won't occur until that class is over and the normal semester starts (when I should have *some* time to work on it again).

**Comment by gr8pefish** [comment]
Closing this issue as I'm cleaning up the issue tracker, and it is a duplicate of 88, as noted above (and can be referenced from it). Will be fixed, just don't need 2 issues for it.
